src/server/pfs/server/driver.go

```diff
@@ -2223,25 +2274,78 @@ func (d *driver) getFile(ctx context.Context, file *pfs.File, offset int64, size
 	if err != nil {
 		return nil, err
 	}
-
 	paths, err := tree.Glob(file.Path)
 	if err != nil {
 		return nil, err
 	}
+	var objects []*pfs.Object
+	var totalSize int64
+	// Add directory path/nodes to unique set of paths
+	// so we can collect and header/footer objects as we traverse the
+	// paths below
 	for path, node := range paths {
-		if node.FileNode == nil {
-			delete(paths, path)
+		if node.FileNode != nil {
+			dirPath := path
+			for dirPath != "/" {
+				dirPath = filepath.Dir(dirPath)
+				if _, ok := paths[dirPath]; ok {
+					continue
+				}
+				dirNode, err := tree.Get(dirPath)
+				if err != nil {
+					return nil, err
+				}
+				paths[dirPath] = dirNode
+			}
+		}
+	}
```

Comment:
Looking forward to when we merge this with @brycemcanally's performance changes this is going to need to be implemented as a single pass over the paths, otherwise it's going to be painfully slow. I don't think it should be too hard to refactor this code to be a single pass. We can chat about how to do it in person.

Reply:
Ok, updated this so that we construct the `objects` array in a single pass, to support a streaming get file operation.

Reply:
:+1: